src/main/kotlin/fi/metatavu/lipsanen/companies/CompanyController.kt

```diff
@@ -0,0 +1,66 @@
+package fi.metatavu.lipsanen.companies
+
+import fi.metatavu.lipsanen.api.model.Company
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.inject.Inject
+import java.util.*
+
+/**
+ * Controller for managing companies
+ */
+@ApplicationScoped
+class CompanyController {
+
+    @Inject
+    lateinit var companyRepository: CompanyRepository
+
+    /**
+     * Lists companies
+     *
+     * @param name company name
+     * @param first first result
+     * @param max max results
+     * @return list of companies
+     */
+    suspend fun list(name: String? = null, first: Int? = null, max: Int? = null): Pair<List<CompanyEntity>, Long> {
+        return companyRepository.list(name = name, first = first, max = max)
+    }
+
+    /**
+     * Creates a new company
+     *
+     * @param company company
+     * @param userId user id
+     * @return created company
+     */
+    suspend fun create(company: Company, userId: UUID): CompanyEntity {
+        return companyRepository.create(
+            id = UUID.randomUUID(),
+            name = company.name,
+            creatorId = userId,
+            lastModifierId = userId
+        )
+    }
+
+    /**
+     * Finds company by id
+     *
+     * @param companyId company id
+     * @return found company or null if not found
+     */
+    suspend fun find(companyId: UUID): CompanyEntity? {
+        return companyRepository.findByIdSuspending(companyId)
+    }
+
+    /**
+     * Updates company
+     *
+     * @param company company
+     * @param foundCompany found company
+     * @param userId user id
+     * @return updated company
+     */
+    suspend fun delete(foundCompany: CompanyEntity) {
+        return companyRepository.deleteSuspending(foundCompany)
+    }
```

Comment:
Do those event function? There are no referenced functions in Company Repository

Reply:
yes, `findByIdSuspending` and `deleteSuspending` come from AbstractRepository which is inherited by CompanyRepository. 
[https://github.com/Metatavu/lipsanen-project-management-api/blob/84019ebc43da082b15ba95310419a05a848d9755/src/main/kotlin/fi/metatavu/lipsanen/persistence/AbstractRepository.kt#L128](url)
[https://github.com/Metatavu/lipsanen-project-management-api/blob/84019ebc43da082b15ba95310419a05a848d9755/src/main/kotlin/fi/metatavu/lipsanen/persistence/AbstractRepository.kt#L62 ](url)